app/api.ts

```diff
@@ -1175,4 +1180,274 @@ export interface RestoredBackup {
   messages: MessageData[]
   participants: Record<string, EntityData>
   hasMoreMessages: boolean
+}
+
+export type MediaData =
+  | EmptyMediaData
+  | PhotoMediaData
+  | GeoMediaData
+  | ContactMediaData
+  | UnsupportedMediaData
+  | DocumentMediaData
+  | WebPageMediaData
+  | VenueMediaData
+  | GameMediaData
+  | InvoiceMediaData
+  | GeoLiveMediaData
+  | PollMediaData
+  | DiceMediaData
+  | StoryMediaData
+  | GiveawayMediaData
+  | GiveawayResultsMediaData
+  | PaidMediaData;
+
+// Define each media type interface
+export interface EmptyMediaData {
+  type: 'empty'
+}
+
+export interface PhotoMediaData {
+  type: 'photo'
+  photo: PhotoData
+  spoiler?: boolean
+  ttlSeconds?: number
+}
+
+export interface GeoPointData {
+  lat: number
+  long: number
+  accessHash: ToString<bigint>
+  accuracyRadius?: number
+}
+
+export interface GeoMediaData {
+  type: 'geo'
+  geo: GeoPointData
+}
+
+export interface ContactMediaData {
+  type: 'contact'
+  phoneNumber: string
+  firstName: string
+  lastName: string
+  vcard: string
+  userId: ToString<EntityID>
+}
+
+export interface UnsupportedMediaData {
+  type: 'unsupported'
+}
+
+export interface DocumentMediaData {
+  type: 'document'
+  document?: DocumentData // as far as I understood, one is for one document and the other is for an array of documents
+  altDocuments?: DocumentData[]
+  nopremium?: boolean
+  spoiler?: boolean
+  video?: boolean
+  round?: boolean
+  voice?: boolean
+  ttlSeconds?: number
+}
+
+// TODO: need to add WebPageData based on TypeWebPage
+export interface WebPageMediaData {
+  type: 'web-page'
+  url: string
+  title?: string
+  description?: string
+  photo?: PhotoData
+}
+
+export interface VenueMediaData {
+  type: 'venue'
+  geo: GeoPointData
+  title: string
+  address: string
+  provider: string
+  venueId: string
```

Comment:
I'm not really sure why but I removed the `Id` suffix from field names in other types. We should probably be consistent or revert that...

Should this be `ToString<bigint>`?

```suggestion
  venue: ToString<bigint>
```